src/errors.rs

```diff
@@ -67,6 +69,12 @@ impl From<jsonwebtoken::errors::Error> for Error {
     }
 }
 
+impl From<yaup::Error> for Error {
+    fn from(error: yaup::Error) -> Error {
+        Error::Yaup(error)
+    }
+}
```

Comment:
Could be cool to add [thiserror](https://docs.rs/thiserror/latest/thiserror/) to the repo in another PR to stop writing this by hand like an animal from the prehistoric era

Reply:
I opened an issue :)